### PR TITLE
OSD Total Distance Flown panel

### PR DESCRIPTION
**Pull request by Hwurzburg**
Replaces previous, closed PR. This only involved the AP_OSD library and cannot impact minimum feature builds like Pixhawk

### CODE REVIEW

File: libraries/AP_OSD/AP_OSD_Screen.cpp [hunk 1]
```diff
@@ -782,6 +788,24 @@ void AP_OSD_Screen::draw_temp(uint8_t x, uint8_t y)
...
 
+void AP_OSD_Screen::draw_dist(uint8_t x, uint8_t y)
+{
+    uint32_t now = AP_HAL::millis();
+    uint32_t delta = now - osd->last_update_ms;
+    AP_AHRS &ahrs = AP::ahrs();
+    Vector2f v = ahrs.groundspeed_vector();
+    float speed = v.length();
+    if (delta > 500) {
+        if (speed < 2.0) speed = 0.0;
```
Comment (sh83) [inline libraries/AP_OSD/AP_OSD_Screen.cpp]: Is this needed?

Comment (Hwurzburg) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: yes...it prevents distance being accumulated while sitting on the ground and ground speed bumps around randomly from 0 up to 2....if IS_FLYING was available to the library functions, that could be the qualifier, but its not apparently...alternatively, I could add an additional storage variable to track flightime and qualify it based on increasing flight time...this was the simplest

Comment (sh83) [reply libraries/AP_OSD/AP_OSD_Screen.cpp]: Ok. It make sense.

**Comment by Hwurzburg** [comment]
Superceded by #9046

**Pull request closed without merging**